pandas/core/internals/blocks.py

```diff
@@ -1248,7 +1248,7 @@ def take_nd(self, indexer, axis, new_mgr_locs=None, fill_tuple=None):
         if fill_tuple is None:
             fill_value = self.fill_value
             new_values = algos.take_nd(values, indexer, axis=axis,
-                                       allow_fill=False)
+                                       allow_fill=False, fill_value=fill_value)
```

Comment:
This modification is deep within the internals (block's definition of `take_nd`), and I am not sure how to really add a test for this.

Reply:
this actually isn't used since allow_fill=False